extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesCommonHelper.java

```diff
@@ -1191,4 +1191,11 @@ private static List<PolicyRule> toPolicyRulesList(Map<String, PolicyRuleConfig>
                         .build())
                 .collect(Collectors.toList());
     }
+
+    private static String parseVCSUri(VCSUriConfig config, ScmInfo scm) {
+        if (config.enabled) {
+            return config.override.orElseGet(() -> scm != null ? Git.sanitizeRemoteUrl(scm.getRemote().get("origin")) : null);
```

Comment:
Mirroring my comment from the original issue. 

Is it reasonable to get ScmInfo directly from SCM configuration compared to SCM configuration of project's build file (e.g. `pom.xml`)? I really wouldn't expect application framework to go digging into my git configuration.

Reply:
SCM configuration may or may not be configured, in contrast to `.git`.
Do you see any reason why the extension should not go digging to the git configuration?

Reply:
Yeah, I really don't understand the argument of not using `.git`...